src/components/ComputeModal.js

```diff
@@ -308,15 +308,22 @@ export const ComputeModalBase = ({
         })
       }
       if (shouldCreateRuntime) {
-        const diskConfig = existingPersistentDisk && !shouldDeletePersistentDisk ? {
-          name: currentPersistentDiskDetails.name
-        } : {
-          name: Utils.generatePersistentDiskName(),
-          size: desiredPersistentDisk.size,
-          diskType: desiredPersistentDisk.diskType.label,
-          labels: { saturnWorkspaceNamespace: namespace, saturnWorkspaceName: name }
-        }
-        const createRuntimeConfig = { ...runtimeConfig, persistentDisk: diskConfig }
+        const diskConfig = Utils.cond(
+          [desiredRuntime.cloudService === cloudServices.DATAPROC, () => { return {} }],
+          [existingPersistentDisk && !shouldDeletePersistentDisk, () => { return { persistentDisk: { name: currentPersistentDiskDetails.name } } }],
```

Comment:
For short arrow functions like this, you can avoid the and `return` by also removing the surrounding `{ }`. When what you want to return is an object literal, put `( )` around it to keep it from being treated as a block statement. I.e.:
```suggestion
          [desiredRuntime.cloudService === cloudServices.DATAPROC, () => ({})],
          [existingPersistentDisk && !shouldDeletePersistentDisk, () => ({ persistentDisk: { name: currentPersistentDiskDetails.name } }) ],
```
Same with the `DEFUALT` case below.